torch/prototypes/nestedtensor/codegen/tensorextension.py

```diff
@@ -0,0 +1,190 @@
+        pass
+
+    if getattr(cls, pbf, False) and not getattr(DefaultClass, pbf, False):
+        raise Exception("WARNING: " + pbf + " already exists "
+                        "and not part of default class")
+
+
+def set_unary_method(cls, tfunc, pbf, inplace):
+    def _gen_func(pbf):
+        def _func(self: cls):
+            assert isinstance(self, cls)
+            if inplace:
+                return getattr(torch, tfunc)(self, out=self)
+            else:
+                return getattr(torch, tfunc)(self)
+        return _func
+    _check_meaningful_overwrite(cls, pbf)
+    setattr(cls, pbf, _gen_func(pbf))
+
+
+def set_binary_method(cls, tfunc, pbf, inplace):
+    def _gen_func(tfunc):
+        def _func(self: cls, other: cls):
+            if inplace:
+                return getattr(torch, tfunc)(self, other, out=self)
+            else:
+                return getattr(torch, tfunc)(self, other)
+        return _func
+    _check_meaningful_overwrite(cls, pbf)
+    setattr(cls, pbf, _gen_func(tfunc))
+
+
+def get_unary_method_signatures():
+    signatures = []
+    for pbf in get_unary_functions():
+        signatures.append({'torch': pbf, 'Tensor': pbf, 'inplace': False})
+        signatures.append({'torch': pbf, 'Tensor': pbf + "_", 'inplace': True})
+    return signatures
+
+
+def get_binary_method_signatures():
+    signatures = []
+    for pbf in ['add', 'mul', 'sub']:
+        signatures.append({'torch': pbf, 'Tensor': pbf, 'inplace': False})
+        signatures.append({'torch': pbf, 'Tensor': pbf + "_", 'inplace': True})
+        signatures.append({'torch': pbf, 'Tensor': "__" + pbf + "__", 'inplace': False})
+        signatures.append({'torch': pbf, 'Tensor': "__i" + pbf + "__", 'inplace': True})
+    signatures.append({'torch': 'div', 'Tensor': 'div', 'inplace': False})
+    signatures.append({'torch': 'div', 'Tensor': 'div' + "_", 'inplace': True})
+    signatures.append({'torch': 'div', 'Tensor': "__" + 'div' + "__", 'inplace': False})
+    signatures.append({'torch': 'div', 'Tensor': "__i" + 'div' + "__", 'inplace': True})
+    return signatures
+
+
+def get_comparison_method_signatures():
+    signatures = []
+    for pbf in get_comparison_functions():
+        signatures.append({'torch': pbf, 'Tensor': pbf, 'inplace': False})
+        signatures.append({'torch': pbf, 'Tensor': pbf + "_", 'inplace': True})
+        signatures.append({'torch': pbf, 'Tensor': "__" + pbf + "__", 'inplace': False})
+    return signatures
+
+
+def add_pointwise_unary_functions(module, cls, func):
+    for function_name in get_unary_functions():
+        set_function(module, cls, function_name, func)
+    for signature in get_unary_method_signatures():
+        set_unary_method(cls, signature['torch'], signature['Tensor'], signature['inplace'])
+    return module, cls
+
+
+def add_pointwise_binary_functions(module, cls, func):
+    for function_name in get_binary_functions():
+        set_function(module, cls, function_name, func)
```

Comment:
It took me a few minutes to figure out what is going on here. `set_function` sets `function_name on `module` to be `partial_apply(func, [function_name, torch.getattr(func)])` if the first argument is `cls`. In that case the naming of this function and its arguments can be improved

Reply:
I'll add more docs